tests/qchem/test_structure.py

```diff
@@ -232,6 +232,36 @@ def test_hf_state(electrons, orbitals, exp_state):
     assert np.allclose(res_state, exp_state)
 
 
+def test_hf_state_basis():
+    r"""Test the correctness of the generated vector in user defined basis"""
+
+    electrons = 2
+    symbols = ["H", "H"]
+    geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4]], requires_grad=False)
```

Comment:
We should use two molecules using `@pytest.mark.parametrize`, suggest `H2` and `H3+`.